pkg/controller/apmserver/config.go

```diff
@@ -66,47 +66,34 @@ func reconcileApmServerConfig(client k8s.Client, as *apmv1.ApmServer) (corev1.Se
 	return reconciler.ReconcileSecret(client, expectedConfigSecret, as)
 }
 
-func NewConfigFromSpec(c k8s.Client, as *apmv1.ApmServer) (*settings.CanonicalConfig, error) {
-	specConfig := as.Spec.Config
-	if specConfig == nil {
-		specConfig = &commonv1.Config{}
+func newConfigFromSpec(c k8s.Client, as *apmv1.ApmServer) (*settings.CanonicalConfig, error) {
+	// Create a base configuration.
```

Comment:
this comment feels like the first step of something bigger that never comes 😄 
I would either remove it or add other comments for other blocks in this function.